### PR TITLE
PMP: Restore array-based soup to mesh conversion

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

This was broken in https://github.com/CGAL/cgal/pull/4418, but untested so unseen.

An example is modified so the code is tested now.

## Release Management

* Affected package(s): `Polygon_mesh_processing`
* Issue(s) solved (if any): -
* Feature/Small Feature (if any): -

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h [hunk 1]
```diff
@@ -37,13 +38,26 @@ namespace CGAL {
...
+{
+  CGAL_static_assertion((std::is_convertible<PS_Point, PM_Point>::value));
+  return PM_Point(p);
+}
+
+// just for backward compatibility reasons
+template <typename PM_Point, typename PS_FT>
+PM_Point convert_to_pm_point(const std::array<PS_FT, 3>& p)
+{
+  return PM_Point(p[0], p[1], p[2]);
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h:52]: why not always using this one?

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/repair_polygon_soup_example.cpp [hunk 2]
```diff
@@ -5,30 +5,55 @@
...
 
-typedef std::vector<std::size_t>                                CGAL_Polygon;
 typedef CGAL::Surface_mesh<Point_3>                             Mesh;
 
+typedef std::array<FT, 3>                                       Custom_point;
+typedef std::vector<std::size_t>                                CGAL_Polygon;
+
 namespace PMP = CGAL::Polygon_mesh_processing;
 
+struct Array_traits
```
Comment (sloriot) [inline Polygon_mesh_processing/examples/Polygon_mesh_processing/repair_polygon_soup_example.cpp:24]: Since a geom traits is needed, I think I would also add in this example the point map. This gives a nice example for a user defined point. You can then undo the changes in the other file.

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h (hunk 1 above)
Comment (MaelRL) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h:52]: * This is weaker than the generic one using explicit conversion;
* `operator[]` is in `CGAL::Point_3` but is not part of its concept;
* Might want to do weird stuff with `CGAL::Point_2` for example.

File: Polygon_mesh_processing/examples/Polygon_mesh_processing/repair_polygon_soup_example.cpp (hunk 2 above)
Comment (MaelRL) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/repair_polygon_soup_example.cpp:24]: But that would go against explicitly testing both functions with `std::array<FT, 3>` as point type, wouldn't it ?

Comment (sloriot) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/repair_polygon_soup_example.cpp:24]: As soon as the geom_traits is needed, the old behavior is broken. But that's not an issue, the old code was somehow incorrect since it was using free functions/operators.

Comment (MaelRL) [reply Polygon_mesh_processing/examples/Polygon_mesh_processing/repair_polygon_soup_example.cpp:24]: The traits is there for `repair_polygon_soup`, not `polygon_soup_to_polygon_mesh`.

**Comment by sloriot** [comment]
Tested in CGAL-5.1-Ic-216
